beets/util/pipeline.py

```diff
@@ -415,8 +446,18 @@ def run_parallel(self, queue_size=DEFAULT_QUEUE_SIZE):
             # Make completely sure that all the threads have finished
             # before we return. They should already be either finished,
             # in normal operation, or aborted, in case of an exception.
-            for thread in threads[:-1]:
-                thread.join()
+            try:
+                for thread in threads[:-1]:
```

Comment:
I'm currently importing some music with that patch, seems fine for now. As long as the threads are non-daemonized, this might not make any difference. But as the last thread (I think) will do the actual writing, it possibly is one of the more important to wait for.